datafusion/expr/src/udf.rs

```diff
@@ -324,26 +328,37 @@ where
     }
 }
 
-/// Trait for implementing [`ScalarUDF`].
+pub struct ScalarFunctionArgs<'a> {
+    // The evaluated arguments to the function
+    pub args: &'a [ColumnarValue],
+    // The number of rows in record batch being evaluated
+    pub number_rows: usize,
+    // The return type of the scalar function returned (from `return_type` or `return_type_from_exprs`)
+    // when creating the physical expression from the logical expression
+    pub return_type: &'a DataType,
```

Comment:
How did I not see this till now? 

Please please please add SessionContext or at least SessionConfig to this. It would allow us to unblock [so](https://github.com/apache/datafusion/issues/13212) [so](https://github.com/apache/datafusion/issues/12892) [many](https://github.com/apache/datafusion/issues/10744) tickets

Reply:
It should also be easy to add as a follow-up?

Reply:
Yes -- I will file a ticket

Reply:
Tracking with https://github.com/apache/datafusion/issues/13519

Reply:
(BTW the change in the design to a struct I think allows for much easier non-breaking API additions in the future)